stdlib/json/0/json.rbs

```diff
@@ -403,3 +403,227 @@ class Array[unchecked out Elem]
+
+  # Returns a hash, that will be turned into a JSON object and represent this
+  # object.
+  #
+  def as_json: (*untyped) -> Hash[String, String | Numeric]
+
+  # Stores class name (Complex) along with real value `r` and imaginary value `i`
+  # as JSON string
+  #
+  def to_json: (?JSON::State state) -> String
+end
+
+class Date
+  # Deserializes JSON string by converting Julian year `y`, month `m`, day `d` and
+  # Day of Calendar Reform `sg` to Date.
+  #
+  def self.json_create: (Hash[String, String | Integer | Float] object) -> instance
+
+  # Returns a hash, that will be turned into a JSON object and represent this
+  # object.
+  #
+  def as_json: (*untyped) -> Hash[String, String | Integer | Float]
+
+  # Stores class name (Date) with Julian year `y`, month `m`, day `d` and Day of
+  # Calendar Reform `sg` as JSON string
+  #
+  def to_json: (?JSON::State state) -> String
+end
+
+class DateTime
+  # Deserializes JSON string by converting year `y`, month `m`, day `d`, hour `H`,
+  # minute `M`, second `S`, offset `of` and Day of Calendar Reform `sg` to
+  # DateTime.
+  #
+  def self.json_create: (Hash[String, String | Integer | Float] object) -> instance
+
+  # Returns a hash, that will be turned into a JSON object and represent this
+  # object.
+  #
+  def as_json: (*untyped) -> Hash[String, String | Integer | Float]
+
+  # Stores class name (DateTime) with Julian year `y`, month `m`, day `d`, hour
+  # `H`, minute `M`, second `S`, offset `of` and Day of Calendar Reform `sg` as
+  # JSON string
+  #
+  def to_json: (?JSON::State state) -> String
+end
+
+class Exception
+  # Deserializes JSON string by constructing new Exception object with message `m`
+  # and backtrace `b` serialized with `to_json`
+  #
+  def self.json_create: (Hash[String, String | Array[String] | nil] object) -> instance
+
+  # Returns a hash, that will be turned into a JSON object and represent this
+  # object.
+  #
+  def as_json: (*untyped) -> Hash[String, String | Array[String] | nil]
+
+  # Stores class name (Exception) with message `m` and backtrace array `b` as JSON
+  # string
+  #
+  def to_json: (?JSON::State state) -> String
+end
+
+class OpenStruct
+  # Deserializes JSON string by constructing new Struct object with values `t`
+  # serialized by `to_json`.
+  #
+  def self.json_create: (Hash[String, String | Hash[Symbol, untyped]] object) -> instance
+
+  # Returns a hash, that will be turned into a JSON object and represent this
+  # object.
+  #
+  def as_json: (*untyped) -> Hash[String, String | Hash[Symbol, untyped]]
+
+  # Stores class name (OpenStruct) with this struct's values `t` as a JSON string.
+  #
+  def to_json: (?JSON::State state) -> String
+end
+
+class Range[out Elem]
+  # Deserializes JSON string by constructing new Range object with arguments `a`
+  # serialized by `to_json`.
+  #
+  def self.json_create: (Hash[String, String | [Elem, Elem, bool]] object) -> instance
```

Comment:
> __[Goodcheck]__ Prefer `boolish` over `bool` for method arguments and block return values ([view](https://sider.review/gh/repos/174796313/pulls/597#issue-5389277))
>
> <details>
>
> See the doc below:
> https://github.com/ruby/rbs/blob/78d04a2db0f1c4925d2b13c2939868edf9465d6c/docs/syntax.md#bool-or-boolish
>
> | Rule |
> | ---- |
> | `rbs.prefer_boolish` |
>
> You can [close this issue](https://sider.review/gh/repos/174796313/pulls/597#issue-5389277) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).
>
> </details>

I closed this issue, because:
- When you strictly want `true | false`.